mosaiic

```diff
@@ -4,25 +4,41 @@ import argparse
 import os
 import sys
 from PIL import Image, ImageDraw, ImageFont, ImageSequence
-
+from multiprocessing import Pool
+from functools import partial
 
 def intensity_to_ascii(intensity: int):
     ascii_palette = " .'`^\",:;Il!i><~+_-?][}{1)(|\\/tfjrxnuvczXYUJCLQ0OZmwqpdbkhao*#MW&8%B@$"
     return ascii_palette[int(intensity/256 * len(ascii_palette))]
 
-
 def ascii_gif(input_gif, font, scale, bg, fg):
     palette = input_gif.getpalette()
     ascii_frames = []
-
+    
     for frame in ImageSequence.Iterator(input_gif):
         frame.putpalette(palette)
         frame_im = Image.new("RGB", frame.size)
         frame_im.paste(frame)
         ascii_frames.append(ascii_image(frame_im, font, scale, bg, fg))
+    return ascii_frames
 
+def ascii_gif_parallel(input_gif, font, scale, bg, fg):
+    palette = input_gif.getpalette()
+    ascii_frames = []    
+    image_iterator = ImageSequence.Iterator(input_gif)
+    func = partial(ascii_image_parallel, font, palette, scale, bg, fg)
+    pool = Pool(8)
```

Comment:
Add a check here so this doesn't exceed the number of cores:
n_cores = 8 if multiprocessing.cpu_count() > 8 else multiprocessing.cpu_count() -1